storey/flow.py

```diff
@@ -608,73 +611,118 @@ async def _handle_completed(self, event, response):
 
 
 class _Batching(Flow):
-    def __init__(self, max_events: Optional[int] = None, timeout_secs=None, **kwargs):
+    def __init__(
+            self,
+            max_events: Optional[int] = None,
+            timeout_secs: Optional[int] = None,
+            group_by_key: bool = False,
+            key: Optional[Union[str, callable]] = None,
```

Comment:
`key_field` or `key_extractor`, really. Do we need `group_by_key` at all? Can't it just be implied when key field is provided?

Reply:
No, because if you want to group by `event.key` you will pass `group_by_key=True` and no `key`

Reply:
Ah. We dealt with this elsewhere using '$key'. What about `group_by_key=False` and `key_field` being defined?

Reply:
> Ah. We dealt with this elsewhere using '$key'. What about **`group_by_key=False` and `key_field`** being defined?

`_Batching` will run as if key_field is not defined.

But i do like the `$key` approach. The only downfall for this is that you have to read the documentation to understand how to use it, whereas switching to `group_by_key=True` will automatically group by `Event.key`. So i guess it's a taste thing. Let me know what is closer to the semantics of the rest of the code base and i'll change it accordingly.